src/futures.rs

```diff
@@ -0,0 +1,18 @@
+//! Asynchronous access to a bincode-encoded item stream using `futures_io`. See the top-level
+//! documentation and the documentation for [`AsyncBincodeReader`], [`AsyncBincodeWriter`], and
+//! [`AsyncBincodeStream`].
```

Comment:
This doesn't seem right. There is no top-level version of `AsyncBincodeReader` or any of the other types any more. I think we should probably place that documentation directly into the generated per-module types.

Reply:
Same goes for the analogous comment in `mod tokio`.

Reply:
"top-level documentation *and* the documentation for" - the macros generate these types, and they exist in the same module as this comment, which is where the links go.

Reply:
I think I've resolved this by expanding the documentation, but let me know what you think.